src/EditorFeatures/Core.Wpf/Options/LegacyEditorConfigDocumentOptionsProvider.DocumentOptions.cs

```diff
@@ -57,6 +58,57 @@ public bool TryGetDocumentOption(OptionKey option, OptionSet underlyingOptions,
                     return false;
                 }
             }
+
+            /// <summary>
+            /// A class that implements <see cref="IReadOnlyDictionary{String, String}" /> atop a <see cref="IReadOnlyDictionary{String, Object}" />
+            /// where we just convert the values to strings with ToString(). Ordering of the underlying dictionary is preserved, so that way
+            /// code that relies on the underlying ordering of the underlying dictionary isn't affected.
+            /// </summary>
+            private class ConvertingDictionary : IReadOnlyDictionary<string, string>
```

Comment:
The intent of a wrapper type here is to be cheaper than allocating a whole new Dictionary and copying everything over. I didn't mind the *old* pattern doing that because we were gaining extra immutability; if we're not doing that (or at least not trying to do it now to be paranoid as a tactical fix), this also may help some perf as well.